src/kernel-install/kernel-install.c

```diff
@@ -168,9 +171,11 @@ static int context_copy(const Context *source, Context *ret) {
         r = strdup_or_null(source->kernel, &copy.kernel);
         if (r < 0)
                 return r;
-        copy.initrds = strv_copy(source->initrds);
-        if (!copy.initrds)
-                return -ENOMEM;
+        if (source->initrds) {
+                copy.initrds = strv_copy(source->initrds);
```

Comment:
Please introduce something like `strv_copy_or_null`.

Reply:
but then NULL would be a valid return value. Means the function would have to return int and write the potential result into a second parameter.